src/libsystemd/sd-netlink/netlink-socket.c

```diff
@@ -238,6 +238,32 @@ int socket_write_message(sd_netlink *nl, sd_netlink_message *m) {
         return k;
 }
 
+int socket_writev_message(sd_netlink *nl, sd_netlink_message *m[], unsigned int msgcount) {
+        struct iovec *iovs;
+        unsigned int i;
+        ssize_t k;
+
+        assert(nl);
+        assert(msgcount);
+
+        iovs = calloc(msgcount, sizeof(struct iovec));
+        if (!iovs)
+                return -errno;
+
+        for (i = 0; i < msgcount; i++) {
+                assert(m[i]->hdr);
+                assert(m[i]->hdr->nlmsg_len);
```

Comment:
we usually try to avoid using C's downgrade-to-bool feature for stuff that is neither pointer nor boolean/boolean-like. i.e. for number checks we usually spell out `!= 0`, so that the reader groks that this is a number, not a boolean.